.github/workflows/ci.yml

```diff
@@ -10,70 +15,91 @@ jobs:
+          echo -e "::group::${{ env.bashInfo }} ${{ env.stepName }} ${{ env.bashEnd }}"
+
           sudo apt-get install -y lcov
           cmake -S test -B build/ \
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
           -DBUILD_CLONE_SUBMODULES=ON \
           -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG'
+
           make -C build/ all
-      - name: Test
-        run: |
-          cd build/
-          ctest -E system --output-on-failure
-          cd ..
-      - name: Run Coverage
+
+          echo "::endgroup::"
+          echo -e "${{ env.bashPass }} ${{env.stepName}} ${{ env.bashEnd }}"
+
+      - name: Run Unit Tests
+        run:  ctest --test-dir build -E system --output-on-failure
+
+      - env:
+          stepName: Line and Branch Coverage Build
         run: |
+          # ${{ env.stepName }}
+          echo -e "::group::${{ env.bashInfo }} Build Coverage Target ${{ env.bashEnd }}"
+          # Build the coverage target
           make -C build/ coverage
-          declare -a EXCLUDE=("\*test\*" "\*CMakeCCompilerId\*" "\*mocks\*")
-          echo ${EXCLUDE[@]} | xargs lcov --rc lcov_branch_coverage=1 -r build/coverage.info -o build/coverage.info
+
+          echo -e "::group::${{ env.bashInfo }} Generate Coverage Report ${{ env.bashEnd }}"
+          # Generate coverage report, excluding extra directories
+          lcov --rc lcov_branch_coverage=1 -r build/coverage.info -o build/coverage.info '*test*' '*CMakeCCompilerId*' '*mocks*'
+
+          echo "::endgroup::"
           lcov --rc lcov_branch_coverage=1 --list build/coverage.info
+          echo -e "${{ env.bashPass }} ${{env.stepName}} ${{ env.bashEnd }}"
       - name: Check Coverage
-        uses: FreeRTOS/CI-CD-Github-Actions/coverage-cop@main
+        uses: FreeRTOS/CI-CD-Github-Actions/coverage-cop@v2
         with:
-          path: ./build/coverage.info
+          coverage-file: ./build/coverage.info
+
   complexity:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Check complexity - Common
-        uses: FreeRTOS/CI-CD-Github-Actions/complexity@main
+        uses: FreeRTOS/CI-CD-Github-Actions/complexity@v2
         with:
           path: ./
+
   doxygen:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Run doxygen build
-        uses: FreeRTOS/CI-CD-Github-Actions/doxygen@main
+        uses: FreeRTOS/CI-CD-Github-Actions/doxygen@v2
         with:
           path: ./
+
   spell-check:
     runs-on: ubuntu-latest
     steps:
       - name: Clone This Repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Run spellings check
-        uses: FreeRTOS/CI-CD-Github-Actions/spellings@main
+        uses: FreeRTOS/CI-CD-GitHub-Actions/spellings@v2
         with:
           path: ./
+
   formatting:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```

Comment:
Why is this ubunto-20.04 while others are running on ubuntu-latest?

Reply:
Uncrustify version 0.69.0_f that is used across all our repos is not the version installed on ubuntu-latest (ubuntu-22.04)